internal/global/trace.go

```diff
@@ -158,6 +158,14 @@ func (t *tracer) Start(ctx context.Context, name string, opts ...trace.SpanStart
 // a nonRecordingSpan by default.
 var autoInstEnabled = new(bool)
 
+// newSpan is called by tracer.Start so auto-instrumentation can attach an eBPF
+// uprobe to this code.
+//
+// Strictly speaking, the following "noinline" flag is not needed, the method's
+// complexity prevents it being inlined. This ensures that is the case going
+// forward by making the restriction explicit.
```

Comment:
I think we can make this a lot shorter.

```suggestion
// "go:noinline" directive prevents the method from ever being inlined.
```

or

```suggestion
// "noinline" pragma prevents the method from ever being inlined.
```

Terms primarily  from: https://github.com/golang/go/blob/069471838998c304ac751b031a19283a9ad30bdb/src/cmd/compile/internal/syntax/syntax.go#L41

Also found here: https://dave.cheney.net/2018/01/08/gos-hidden-pragmas